src/main/webapp/app/course/competencies/competency-management/competency-management-table.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output, inject } from '@angular/core';
+import { CompetencyService } from 'app/course/competencies/competency.service';
+import { AlertService } from 'app/core/util/alert.service';
+import { CompetencyRelation, CompetencyRelationDTO, CompetencyWithTailRelationDTO, CourseCompetency, dtoToCompetencyRelation, getIcon } from 'app/entities/competency.model';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { filter, map } from 'rxjs/operators';
+import { onError } from 'app/shared/util/global.utils';
+import { Subject } from 'rxjs';
+import { faFileImport, faPencilAlt, faPlus, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ImportAllCompetenciesComponent, ImportAllFromCourseResult } from 'app/course/competencies/competency-management/import-all-competencies.component';
+import { PrerequisiteService } from 'app/course/competencies/prerequisite.service';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisMarkdownModule } from 'app/shared/markdown.module';
+
+@Component({
+    selector: 'jhi-competency-management-table',
+    templateUrl: './competency-management-table.component.html',
+    standalone: true,
+    imports: [ArtemisSharedModule, ArtemisMarkdownModule],
+})
+export class CompetencyManagementTableComponent implements OnInit, OnDestroy {
+    @Input() courseId: number;
+    @Input() courseCompetencies: CourseCompetency[];
+    @Input() relations: CompetencyRelation[];
+    @Input() competencyType: 'competency' | 'prerequisite';
+    @Input() standardizedCompetenciesEnabled: boolean;
+
+    @Output() competencyDeleted = new EventEmitter<number>();
+
+    service: CompetencyService | PrerequisiteService;
```

Comment:
Can we not just use the base type CourseCompetencyService here?

Reply:
No, since these two Services offer more methods than the CourseCompetencyService. We could introduce an interface that the two service implement though 🤔 
Also general question: What do you think about the inheritance of CompetencyService and PrerequisiteService from CourseCompetencyService? Should we keep it?

Reply:
I saw that the services implement different methods but in this component only the functionalities from the extended class CourseCompetencyService are used so we could just use the base class as an input here.
In general I think it's fine to have  to inherit of courseCompetencyService for now.

Reply:
Actually not, since only the subclasses offer a delete functionality